frontends/web/src/components/dropdown/mobile-fullscreen-selector.tsx

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright 2025 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ReactNode, useState, useEffect } from 'react';
+import { ActionMeta } from 'react-select';
+import { TOption } from './dropdown';
+import { ChevronLeftDark } from '@/components/icon';
+import { useTranslation } from 'react-i18next';
+import styles from './mobile-fullscreen-selector.module.css';
+
+type Props<T, IsMulti extends boolean = false> = {
+  title?: string;
+  options: TOption<T>[];
+  renderOptions: (option: TOption<T>) => ReactNode;
+  value: IsMulti extends true ? TOption<T>[] : TOption<T>;
+  onChange: (newValue: IsMulti extends true ? TOption<T>[] : TOption<T>, actionMeta: ActionMeta<TOption<T>>) => void;
+  isMulti?: boolean;
+  isOpen?: boolean;
+  onOpenChange?: (isOpen: boolean) => void;
+}
+
+export const MobileFullscreenSelector = <T, IsMulti extends boolean = false>({
+  title = 'Select',
+  options,
+  renderOptions,
+  value,
+  onChange,
+  isMulti,
+  isOpen: controlledIsOpen,
+  onOpenChange,
+}: Props<T, IsMulti>) => {
+  const [internalIsOpen, setInternalIsOpen] = useState(false);
+  const [searchText, setSearchText] = useState('');
+  const { t } = useTranslation();
+
+  const isControlled = controlledIsOpen !== undefined && onOpenChange !== undefined;
+  const isOpen = isControlled ? controlledIsOpen : internalIsOpen;
+
+  useEffect(() => {
+    if (!isOpen) {
+      setSearchText('');
+    }
+  }, [isOpen]);
+
+  const handleOpen = () => {
+    if (isControlled && onOpenChange) {
```

Comment:
`isControlled` already contains `onOpenChange`, so it looks like it is not needed to check for `onOpenChange` again.

small nit: I find it a bit hard to read:  `isOpen` arg, `isOpen` const, `controlledIsOpen`, `internalIsOpen`, `isControlled` maybe this could be slightly simplified? But if too much hassle leave as is.